apps/server/src/shared/repo/system/system.repo.integration.spec.ts

```diff
@@ -25,4 +28,36 @@ describe('system repo', () => {
 		expect(repo).toBeDefined();
 		expect(typeof repo.findById).toEqual('function');
 	});
+
+	describe('findById', () => {
+		afterEach(async () => {
+			await em.nativeDelete(System, {});
+		});
+
+		it('should return right keys', async () => {
+			const system = systemFactory.build();
+			await em.persistAndFlush([system]);
+			const result = await repo.findById(system.id);
+			expect(Object.keys(result).sort()).toEqual(
+				['createdAt', 'updatedAt', 'type', 'url', 'alias', 'oauthconfig', '_id'].sort()
+			);
+		});
+
+		it('should return a System that matched by id', async () => {
+			const system = systemFactory.build();
+			await em.persistAndFlush([system]);
+			const result = await repo.findById(system.id);
+			expect(result).toEqual(system);
+		});
+
+		it('should throw an error if System by id doesnt exist', async () => {
+			const idA = new ObjectId().toHexString();
+
+			await expect(repo.findById(idA)).rejects.toThrow(NotFoundError);
+		});
+
+		// it() -> design test that does something with a Systementity, so that we know, that System has been loaded correctly
```

Comment:
the tests above should be enough for such a simple get function.
You dont need to assume that your data is broken or mikroorm messes up mapping, beyond making sure that the data is mapped correctly (as you ensured above), and the result is indeed a system entity (as also ensured above).

Everything else is responsability of either the entity and the components working with it (eg ensuring it can only be constructed with valid data), or the mikroORM itself.